api/app/Models/Pool.php

```diff
@@ -63,4 +67,32 @@ public function poolCandidates(): HasMany
     {
         return $this->hasMany(PoolCandidate::class);
     }
+
+    /* accessor to obtain Advertisement Status, depends on two variables regarding published and expiry */
+    public function getAdvertisementStatusAccessorAttribute()
```

Comment:
I think if this function is named correctly you will get a field in the schema just named `AdvertisementStatus` that you can query.  That would be much more natural.